src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -88,6 +88,30 @@ public SchemaObjectType(ObjectSchema objectSchema, BuildContext context) : base(
             }
         }
 
+        public bool RequiresFactoryMethod
+        {
+            get
+            {
+                if (Declaration.Accessibility != "public" || !IncludeDeserializer)
```

Comment:
Do we need the `!IncludeDeserializer` check?

Reply:
I wasn't able to ensure that public read-only model always has constructor for deserialization, so I'd rather not generate a factory than produce code that wouldn't compile.
As an alternative, we can enforce those constructors on public read-only models.

Reply:
Your next condition should handle all theoretically existing cases and if it's not I would like to know which ones did we miss. I'm afraid that checking `IncludeDeserializer` can hide some cases that needed a factory but didn't get one.

Reply:
Ok

Reply:
Fixed